src/Compilers/VisualBasic/Portable/Compilation/SyntaxTreeSemanticModel.vb

```diff
@@ -863,8 +863,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         ''' inside the syntax span. Returns Nothing otherwise.
         ''' </summary>
         Private Function CheckSymbolLocationsAgainstSyntax(symbol As NamedTypeSymbol, nodeToCheck As VisualBasicSyntaxNode) As NamedTypeSymbol
-            For Each location In symbol.Locations
-                If location.SourceTree Is Me.SyntaxTree AndAlso nodeToCheck.Span.Contains(location.SourceSpan.Start) Then
+            For Each reference In symbol.DeclaringSyntaxReferences
+                If reference.SyntaxTree Is Me.SyntaxTree AndAlso nodeToCheck.Span.Contains(reference.Span.Start) Then
                     Return symbol
```

Comment:
Originally, the symbols were filtered based on the location, which is actually coming from the type name. This doesn't work when the user is in the process of declaring the type because you get something like this:
```
Partial Class $$
```
The location of the name is where the cursor is placed, which also happens to be the exact end of the span. Since the span is right-open interval, the intersection checks fails. Basing the check on the declaration would seem to be more reliable and in line with the intent of the callers.

That said, this change makes me nervous. Based only following some call paths, it looks like it will have a fairly broad impact, and I haven't found any tests which directly validate intended behaviour. I'd appreciate some feedback here.